### PR TITLE
teach typescript sdk that not all entities have an artificial primary…

### PR DESCRIPTION
**Pull request by DanielCollins**
… key

named id.

### CODE REVIEW

File: typescript/src/entity.ts [hunk 1]
```diff
@@ -248,10 +260,10 @@ export class Entity {
     Object.defineProperties(this, properties);
   }
 
-  public static get<T extends typeof Entity>(this: T, id: number,
+  public static get<T extends typeof Entity>(this: T, key: number,
```
Comment (s3341458) [inline typescript/src/entity.ts]: just for curiosity,  key must be number or String right?

File: typescript/src/entity.ts [hunk 2]
```diff
@@ -437,7 +449,7 @@ export class Entity {
   }
 
   public save = () => {
-    const primaryKey: number = (this as any).id;
+    const primaryKey: number = this.getPrimaryKeyValue();
```
Comment (s3341458) [inline typescript/src/entity.ts]: Do not sure about how typescript works, for now I can use my knowledge about JAVA to infer it.

just the  same question as above, primaryKeyValue can be string for token

File: typescript/src/entity.ts [hunk 3]
```diff
@@ -636,7 +648,7 @@ export class Entity {
   }
 
   public delete = () => {
-    const primaryKey: number = (this as any).id;
+    const primaryKey: number = this.getPrimaryKeyValue();
```
Comment (s3341458) [inline typescript/src/entity.ts]: as above

**Comment by DanielCollins** [comment]
well, more generally, it could be an array of scalars, nothing says that a primary key is a single column, but i don't want to support that right now.

**Pull request merged**